core/src/main/java/org/elasticsearch/common/network/NetworkService.java

```diff
@@ -273,4 +270,19 @@ public InetAddress resolvePublishHostAddresses(String publishHosts[]) throws IOE
         }
         return InetAddress.getAllByName(host);
     }
+
+    /**
+     * Register custom name resolver a DiscoveryPlugin might provide
+     * @param discoveryPlugins Discovery plugins
+     */
+    public static List<CustomNameResolver> registerCustomNameResolvers(Settings settings, List<DiscoveryPlugin> discoveryPlugins) {
```

Comment:
I don't think it should be in `DiscoveryPlugin`, since it is an implementation detail. Why is it used in the transport client? I did not think any of our cloud plugins were "transport client plugins"? (note that we currently do not have them marked for publication to maven, like we added for eg reindex)